cpp/src/parquet/reader_test.cc

```diff
@@ -127,6 +130,85 @@ void CheckRowGroupMetadata(const RowGroupMetaData* rg_metadata,
   }
 }
 
+class TestTextDeltaLengthByteArray : public ::testing::Test {
+ public:
+  void SetUp() { reader_ = ParquetFileReader::OpenFile(text_delta_len_byte_array()); }
+
+  void TearDown() {}
+
+ protected:
+  std::unique_ptr<ParquetFileReader> reader_;
+};
+
+TEST_F(TestTextDeltaLengthByteArray, TestTextScanner) {
+  std::shared_ptr<RowGroupReader> group = reader_->RowGroup(0);
+
+  // column 0, id
+  std::shared_ptr<ByteArrayScanner> scanner(new ByteArrayScanner(group->Column(0)));
+  ByteArray val;
+  bool is_null;
+  std::string expected_prefix("apple_banana_mango");
+  for (int i = 0; i < 1000; ++i) {
+    ASSERT_TRUE(scanner->HasNext());
+    ASSERT_TRUE(scanner->NextValue(&val, &is_null));
+    ASSERT_FALSE(is_null);
+    std::string expected = expected_prefix + std::to_string(i * i);
+    ASSERT_TRUE(val.len == expected.length());
+    ASSERT_EQ(std::memcmp(val.ptr, expected.c_str(), val.len), 0);
+  }
+  ASSERT_FALSE(scanner->HasNext());
+  ASSERT_FALSE(scanner->NextValue(&val, &is_null));
+}
+
+TEST_F(TestTextDeltaLengthByteArray, TestBatchRead) {
+  std::shared_ptr<RowGroupReader> group = reader_->RowGroup(0);
+
+  // column 0, id
+  std::shared_ptr<ByteArrayReader> col =
+      std::dynamic_pointer_cast<ByteArrayReader>(group->Column(0));
+
+  int16_t def_levels[25];
```

Comment:
Could you add `int64_t batch_size = 25;` and use it instead of putting many `25` literals?